lib/cocoapods-core/dependency.rb

```diff
@@ -249,24 +249,41 @@ def merge(other)
       unless name == other.name
         raise ArgumentError, "#{self} and #{other} have different names"
       end
+
       default   = Requirement.default
       self_req  = requirement
       other_req = other.requirement
 
-      dep = if other_req == default
-              self.class.new(name, self_req)
+      req = if other_req == default
+              self_req
             elsif self_req == default
-              self.class.new(name, other_req)
+              other_req
             else
-              self.class.new(name, self_req.as_list.concat(other_req.as_list))
+              self_req.as_list.concat(other_req.as_list)
             end
 
+      opts = {}
+
       if external_source || other.external_source
-        self_external_source  = external_source || {}
-        other_external_source = other.external_source || {}
-        dep.external_source = self_external_source.merge(other_external_source)
+        opts.
+          merge!(external_source || {}) { raise ArguementError, 'key conflict' }.
```

Comment:
This exception isn't helpful, should update to include which key conflicts and what their values are.

Reply:
Also `ArgumentError`